src/eureka/S5_lightcurve_fitting/differentiable_models/StarryModel.py

```diff
@@ -113,99 +113,65 @@ def setup(self, newparams):
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    value = getattr(self.model, item0)
-                    setattr(spotrad[chan, n], item0, value)
-                    # read latitudes
-                    item0 = 'spotlat' + str(n)
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    setattr(spotlat[chan, n], item0, value)
-                    # read longitudes
-                    item0 = 'spotlon' + str(n)
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    setattr(spotlon[chan, n], item0, value)
-                    # read contrasts
-                    item0 = 'spotcon0'
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    setattr(spotcon[chan, n], item0, value)
-                # read number of points
-                starres = self.parameters.dict['spotnpts'][0]
-                # read stellar inclination (if provided)
-                if 'spotstari' in self.parameters.dict.keys():
-                    item0 = 'spotstari'
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    setattr(starrot[chan], item0, value)
-                # read stellar rotation (if provided)
-                if 'spotrot' in self.parameters.dict.keys():
-                    item0 = 'spotrot'
-                    if self.parameters.dict[item0][1] == 'free':
-                        item0 += channel_id
-                    setattr(starrot[inc], item0, value)
+                spotrad = tt.zeros(0)
+                spotlat = tt.zeros(0)
+                spotlon = tt.zeros(0)
+                spotcon = tt.zeros(0)
+
+                for n in range(pl_params.nspots):
+                    # read radii, latitudes, longitudes, and contrasts
+                    if n > 0:
+                        spot_id = f'{n}'
+                    else:
+                        spot_id = ''
+                    spotrad = tt.concatenate([
+                        spotrad, [getattr(pl_params, f'spotrad{spot_id}'),]])
+                    spotlat = tt.concatenate([
+                        spotlat, [getattr(pl_params, f'spotlat{spot_id}'),]])
+                    spotlon = tt.concatenate([
+                        spotlon, [getattr(pl_params, f'spotlon{spot_id}'),]])
+                    spotcon = tt.concatenate([
+                        spotcon, [getattr(pl_params, f'spotcon{spot_id}'),]])
+
+                # Apply some conversions since inputs are in fleck units
+                spotrad *= 90
+                spotcon = 1-spotcon
+
+                if pl_params.spotnpts is None:
+                    # Have a default spotnpts for starry
+                    pl_params.spotnpts = 30
```

Comment:
Should we assign a default value here or in AstroModel.py?

Reply:
AstroModel.py can't tell whether you're fitting with Fleck or Starry, which is why I put these package-specific default values in their respective model files